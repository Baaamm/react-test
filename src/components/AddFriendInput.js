import React from 'react';
import { PropTypes } from 'prop-types'
import classnames from 'classnames';
import styles from './AddFriendInput.css';

const AddFriendInput = ( {addFriend} ) => (
    <input
        type="text"
        autoFocus="true"
        className={classnames('form-control', styles.addFriendInput)}
        placeholder="Type the name of a friend"
        onKeyDown={(e) => {
                if(e.which===13) {
                    addFriend(e.target.value);
                    e.target.value = '';
                }
            }
        }
    />
);

AddFriendInput.propTypes = {
    addFriend: PropTypes.func.isRequired
};

export default AddFriendInput
