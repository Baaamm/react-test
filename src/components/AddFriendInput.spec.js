import React from 'react';
import AddFriendInput from './AddFriendInput';
import renderer from 'react-test-renderer';

it('renders only AddFriendInput correct', () => {
    const tree = renderer.create(<AddFriendInput addFriend = {() => {} }/>).toJSON();
    expect(tree).toMatchSnapshot();
});
