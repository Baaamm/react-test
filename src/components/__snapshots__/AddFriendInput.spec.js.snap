exports[`test renders only AddFriendInput correct 1`] = `
<input
  autoFocus="true"
  className="form-control"
  onKeyDown={[Function]}
  placeholder="Type the name of a friend"
  type="text" />
`;
